From 115b41568d368a4bcd716407076716be9fbf2f43 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 7 Mar 2016 05:18:14 +0700
Subject: [PATCH 43/54] 
 0059-Fix-video-audio-sync-issue-when-audio-track-is-chang.patch

Change-Id: I020e94366219565bfe4f72290c8fc0b3de17f035
---
 media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index 8b7690d..e5495fe 100755
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -1367,6 +1367,10 @@ void NuPlayer::Renderer::onResume() {
 
     if (!mAudioQueue.empty()) {
         postDrainAudioQueue_l();
+    } else if (mHasAudio) {
+        // Audio decoder is not ready when resuming playback after
+        // changing audio track. Queue synching is needed.
+        mSyncQueues = true;
     }
 
     if (!mVideoQueue.empty()) {
-- 
2.5.0

